final/src/main/java/ua/epam/spring/hometask/service/impl/BookingServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package ua.epam.spring.hometask.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ua.epam.spring.hometask.dao.AuditoriumDAO;
+import ua.epam.spring.hometask.dao.TicketDAO;
+import ua.epam.spring.hometask.domain.*;
+import ua.epam.spring.hometask.service.BookingService;
+import ua.epam.spring.hometask.service.DiscountService;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Set;
+
+@Service
+public class BookingServiceImpl implements BookingService {
+
+    @Autowired
+    private TicketDAO ticketDAO;
+
+    @Autowired
+    private AuditoriumDAO auditoriumDAO;
+
+    @Autowired
+    private DiscountService discountService;
+
+    private double vipSeatCoefficient;
+
+    private static Map<EventRating, Double> ratingCoefficients;
+
+    @Override
+    public double getTicketsPrice(@Nonnull Event event, @Nonnull LocalDateTime dateTime,
+                                  @Nullable User user, @Nonnull Set<Long> seats) {
+        int countSeats = seats.size();
+
+        Auditorium auditorium = auditoriumDAO.findAuditoriumOnDateTime(event.getAuditoriums(), dateTime);
+
+        if (auditorium == null) {
+            throw new IllegalArgumentException("There is no auditorium for particular date and time.");
+        }
+
+        double eventPrice = calculateEventPrice(event, auditorium, seats, countSeats);
+        double eventDiscountPercent = discountService.getDiscount(user, event, dateTime, countSeats);
```

Comment:
Both calculateEnventPrice and discountService.getDiscount invocations can be called inside calculateEventPriceByDiscountPercent method.